#### Hypothesis 3: There has been an increase in discount provided across products, contributing to lower Median Order Value and inconsistent revenue growth.

In [1]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt

In [2]:
# loading the data
data = pd.read_parquet("../../data/processed/superstore/data.parquet")
orders = pd.read_parquet("../../data/processed/superstore/orders.parquet")
orders_agg = pd.read_parquet("../../data/processed/superstore/orders_agg.parquet")
dates = pd.read_parquet("../../data/processed/superstore/dates.parquet")
product = pd.read_parquet("../../data/processed/superstore/product.parquet")
product_price_disc = pd.read_parquet("../../data/processed/superstore/product_price_disc.parquet")

#### Overall discount trend

In [26]:
# determing whether the typical discount offered has changed over time
data[["Order ID","Discount"]].merge(
    orders[["Order ID","Order Date"]]
).merge(
    dates[["date","year"]],
    left_on="Order Date",
    right_on="date"
    ).groupby("year")["Discount"].agg(
        ["mean","median"]
    ).round(2)

,mean,median
year,,
2014,0.15,0.20
2015,0.16,0.20
2016,0.15,0.00
2017,0.15,0.15


- The typical discount value increased after 2015. It went from ~0.31 to ~0.31.5
- The discount increased by bit, the revenue growth slowed. Therefore, potentially reducing the revenue generated from each order and contributing to declining MOV

#### Discount by Category

In [44]:
# discount by Category - MEAN
data[["Order ID","Discount"]].merge(
    orders[["Order ID","Order Date","Product ID"]]
).merge(
    dates[["date","year"]],
    left_on="Order Date",
    right_on="date"
    ).merge(
    product[["Product ID", "Category"]],
    left_on="Product ID", right_on="Product ID"
    ).groupby(
        ["year","Category"]
        )["Discount"].agg(
            ['mean','median']
).reset_index().pivot(
    columns="year",
    index="Category",
    values="mean"
).round(2)

year,2014,2015,2016,2017
Category,,,,
Furniture,0.16,0.16,0.17,0.16
Office Supplies,0.15,0.16,0.14,0.15
Technology,0.15,0.15,0.15,0.14


In [43]:
# discount by Category - MEDIAN
data[["Order ID","Discount"]].merge(
    orders[["Order ID","Order Date","Product ID"]]
).merge(
    dates[["date","year"]],
    left_on="Order Date",
    right_on="date"
    ).merge(
    product[["Product ID", "Category"]],
    left_on="Product ID", right_on="Product ID"
    ).groupby(
        ["year","Category"]
        )["Discount"].agg(
            ['mean','median']
).reset_index().pivot(
    columns="year",
    index="Category",
    values="median"
).round(2)

year,2014,2015,2016,2017
Category,,,,
Furniture,0.20,0.20,0.2,0.2
Office Supplies,0.15,0.15,0.0,0.0
Technology,0.20,0.20,0.2,0.2


##### Furniture
- Median discount remained cosnsitent as of 205 throught out the analysis period.
- Mean discount fluctuated between 16% and 17%
- Both measures indicate that the discount distribution for furniture remained highly stable over time, with minimal year-to-year variation
- The company's pricing strategy for Furniture remained consistent throughout the four-year period. 
    - Since both the typical discount and the average discount changed very little, it is unlikely that discounting played a significant role in changes to Furniture's revenue performance.

##### Office supplies
- Median discount decreased from 15% to 0% after 2015.
- Mean discount remained close to 15%.
- The decline in the median indicates that the typical Office Supplies transaction became less likely to receive a discount after 2015. 
    - However, the relatively stable mean suggests that higher discounts were still applied to a smaller subset of transactions, keeping the overall average discount nearly unchanged.
- It suggests the company became more selective about discounting.
    - Instead of giving moderate discounts to many customers, it likely offered larger discounts to a smaller group of transactions while selling most orders at full price.

##### Technology
- Median discount remained constant at 20%.
- Mean discount stayed around 14–15%.
- Both statistics show very little variation, indicating a stable discounting pattern throughout the analysis period.
- Technology maintained a very consistent discounting strategy throughout the analysis period.



#### Discount Distribution

In [25]:
# were discounts more frequent or simply became larger
data[["Order ID","Discount"]].merge(
    orders[["Order ID","Order Date"]]
).merge(
    dates[["date","year"]],
    left_on="Order Date",
    right_on="date"
    ).groupby("year")["Discount"].describe().round(2)

,count,mean,std,min,25%,50%,75%,max
year,,,,,,,,
2014,6005.0,0.15,0.20,0.0,0.0,0.20,0.2,0.8
2015,6372.0,0.16,0.21,0.0,0.0,0.20,0.2,0.8
2016,7705.0,0.15,0.20,0.0,0.0,0.00,0.2,0.8
2017,9820.0,0.15,0.20,0.0,0.0,0.15,0.2,0.8


- Mean: The mean discount remained between 15% and 16% throughout the four-year period.
    - The average level of discounting was highly stable with negligible year-to-year variation.
-SD: The standard deviation remained approximately 0.20 each year.
    - The variability of discounts remained consistent, suggesting that the spread of discount values did not materially change over time.
- Quartiles: 
    - 25th Percentile = 0%:
        - At least 25% of all transactions received no discount.
    - Median = 50%:
        - The typical discount offered changed slightly in 2016 and 2017, indicating that a greater proportion of transactions received little or no discount compared with earlier years.
    - 75th Percentile = 20%:
        - At least 75% of all transactions received discounts of 20% or less.
- Maximum: Maximum discount remained 80% every year.
    - The highest promotional discounts offered by the company did not change throughout the analysis period.


Things to note:
- The distribution is quite stable. Nothing suggests that the company suddenly became more aggresive with discounting

Therefore, the company's discounting strategy remained largely consistent throughout the analysis period. 
Although, the typical discount changed modestly in 2016 & 2017, the overall average discount, discount variability, upper quartiles and maximum discount remained stable. 

These findings suggest that the business did not rely on increasinlgy aggressive discounting to drive sales.

#### Revenue vs Discount

Discount vs Median Order Value

Correlation